### PR TITLE
support setting charging rule in PCC by webconsole

### PR DESCRIPTION
**Pull request by Roy-Hu**

### CODE REVIEW

File: internal/sbi/producer/smpolicy.go [hunk 1]
```diff
@@ -222,17 +224,116 @@ func createSMPolicyProcedure(request models.SmPolicyContextData) (
 		logger.SmPolicyLog.Errorf("createSMPolicyProcedure error: %+v", err)
 	}
 
+	filterCharging := bson.M{"ueId": ue.Supi, "snssai": util.SnssaiModelsToHex(*request.SliceInfo), "dnn": "", "filter": ""}
```
Comment (ianchen0119) [inline internal/sbi/producer/smpolicy.go:227]:
```go
filterCharging := bson.M{"ueId": ue.Supi, "snssai": util.SnssaiModelsToHex(*request.SliceInfo)}
```

File: internal/sbi/producer/smpolicy.go [hunk 2]
```diff
@@ -222,17 +224,116 @@ func createSMPolicyProcedure(request models.SmPolicyContextData) (
...
+			}
+
+			pccRule := util.CreatePccRule(smPolicyData.PccRuleIdGenerator, 33, []models.FlowInformation{
+				{
+					FlowDescription: FlowDespcriptionStr,
+					FlowDirection:   models.FlowDirectionRm_DOWNLINK,
+				},
+			}, "")
+			// qfi := strconv.Itoa(int(flowRule["qfi"].(float64)))
+			// util.SetPccRuleRelatedByQFI(&decision, pccRule, qfi)
```
Comment (ianchen0119) [inline internal/sbi/producer/smpolicy.go:296]: Why comment the QFI operations?

File: internal/sbi/producer/smpolicy.go [hunk 3]
```diff
@@ -222,17 +224,116 @@ func createSMPolicyProcedure(request models.SmPolicyContextData) (
 		logger.SmPolicyLog.Errorf("createSMPolicyProcedure error: %+v", err)
 	}
 
+	filterCharging := bson.M{"ueId": ue.Supi, "snssai": util.SnssaiModelsToHex(*request.SliceInfo), "dnn": "", "filter": ""}
+	chargingInterface, err := mongoapi.RestfulAPIGetOne(chargingDataColl, filterCharging)
+	if err != nil {
+		logger.SmPolicyLog.Errorf("Fail to get charging data to mongoDB err: %+v", err)
+	}
+	logger.SmPolicyLog.Errorf("chargingInterface %+v", chargingInterface)
```
Comment (ianchen0119) [inline internal/sbi/producer/smpolicy.go:232]: remove this line

File: internal/sbi/producer/smpolicy.go [hunk 4]
```diff
@@ -222,17 +224,125 @@ func createSMPolicyProcedure(request models.SmPolicyContextData) (
...
 	}
 
+	filterCharging := bson.M{"ueId": ue.Supi, "snssai": util.SnssaiModelsToHex(*request.SliceInfo)}
+	chargingInterface, err := mongoapi.RestfulAPIGetOne(chargingDataColl, filterCharging)
+	if err != nil {
+		logger.SmPolicyLog.Errorf("Fail to get charging data to mongoDB err: %+v", err)
+		logger.SmPolicyLog.Errorf("chargingInterface %+v", chargingInterface)
+	}
+
+	pcc := util.CreateDefaultPccRules(smPolicyData.PccRuleIdGenerator)
```
Comment (tim-ywliu) [inline internal/sbi/producer/smpolicy.go:234]: should handle charging when err is nil?

File: internal/sbi/producer/smpolicy.go [hunk 5]
```diff
@@ -222,17 +224,126 @@ func createSMPolicyProcedure(request models.SmPolicyContextData) (
...
+	smPolicyData.PccRuleIdGenerator++
+
+	filterCharging := bson.M{"ueId": ue.Supi, "snssai": util.SnssaiModelsToHex(*request.SliceInfo)}
+	chargingInterface, err := mongoapi.RestfulAPIGetOne(chargingDataColl, filterCharging)
+	if err != nil {
+		logger.SmPolicyLog.Errorf("Fail to get charging data to mongoDB err: %+v", err)
+		logger.SmPolicyLog.Errorf("chargingInterface %+v", chargingInterface)
+		util.SetPccRuleRelatedData(&decision, pcc, nil, nil, nil, nil)
+	} else {
```
Comment (tim-ywliu) [inline internal/sbi/producer/smpolicy.go:236]: @ianchen0119 
`} else if chargingInterface != nil {`

File: internal/context/ue.go [hunk 6]
```diff
@@ -156,6 +157,8 @@ func (ue *UeContext) NewUeSmPolicyData(
 	// data.RefToAmPolicy = amData
 	data.PccRuleIdGenerator = 1
 	data.ChargingIdGenerator = 1
+	// data.RatingGroupIdGenerator = 1
```
Comment (ianchen0119) [inline internal/context/ue.go:160]: Remove the commented line.

**Review by ianchen0119 — APPROVED** [review]

**Review by tim-ywliu — APPROVED** [review]

**Pull request merged**